src/ORM/Rule/ExistsIn.php

```diff
@@ -85,25 +97,37 @@ public function __invoke(EntityInterface $entity, array $options)
             return true;
         }
 
-        $nulls = 0;
-        $schema = $source->schema();
-        foreach ($this->_fields as $field) {
-            if ($schema->column($field) && $schema->isNullable($field) && $entity->get($field) === null) {
-                $nulls++;
-            }
-        }
-        if ($nulls === count($this->_fields)) {
+        if ($this->_fieldsAreNull($entity, $source)) {
             return true;
         }
 
         $primary = array_map(
-            [$this->_repository, 'aliasField'],
+            [$target, 'aliasField'],
             $bindingKey
         );
         $conditions = array_combine(
             $primary,
             $entity->extract($this->_fields)
         );
-        return $this->_repository->exists($conditions);
+        return $target->exists($conditions);
+    }
+
+    /**
+     * Check whether or not the entity fields are null and nullable.
```

Comment:
Logically they are first nullable and then also null, but nitpick. No big deal :)